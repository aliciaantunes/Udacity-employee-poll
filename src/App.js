import React from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { getQuestionsHandler, getUsersHandler } from './redux/action-handlers/questionsActions';

function App() {
  const dispatch = useDispatch();
  const questions = useSelector(state => state.questions);
  const users = useSelector(state => state.users);

  const fetchQuestions = () => {
    console.log('Fetching questions...');
    dispatch(getQuestionsHandler());
  };

  const fetchUsers = () => {
    console.log('Fetching users...');
    dispatch(getUsersHandler());
  };

 console.log('questions:', questions);
 console.log('users:', users);
  return (
    <div className="App">
      <header className="App-header">
      <button onClick={fetchQuestions}>Fetch Questions</button>
      <button onClick={fetchUsers}>Fetch Users</button>
        <ul>
          {questions.map(question => (
            <li key={question.id}>{question.optionOne.text}</li>
          ))}
        </ul>
        <ul>
          {users.map(user => (
            <li key={user.id}>{user.name}</li>
          ))}
        </ul>
      </header>
    </div>
  );
}

export default App;
