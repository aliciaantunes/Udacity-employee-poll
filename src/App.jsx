import React from "react";
import { useDispatch, useSelector } from "react-redux";
import {
  getQuestionsHandler,
  getUsersHandler,
  addQuestionHandler,
} from "./redux/action-handlers/questionsActions";

// Redux flow:
// 1. User click in the button to fetch
// 2. The fetchQuestions function is called
// 3. The fetchQuestions function dispatch the getQuestionsHandler action
// 4. The getQuestionsHandler fetch the data from the Fake API
// 5. The getQuestionsHandler dispatch the reducer with the data
// 6. The reducer update the state with the new data
// 7. The useSelector hook get the new data from the state

function App() {
  const dispatch = useDispatch();
  const questions = useSelector((state) => state.questions);
  const users = useSelector((state) => state.users);

  const fetchQuestions = () => {
    console.log("Fetching questions...");
    dispatch(getQuestionsHandler());
  };

  const fetchUsers = () => {
    console.log("Fetching users...");
    dispatch(getUsersHandler());
  };

  const newQuestion = () => {
    console.log("Creating new question...");

    const question = {
      author: "tylermcginnis",
      optionOneText: "Mexerica",
      optionTwoText: "Pokan",
    };

    dispatch(addQuestionHandler(question));
  };

  // const newAnswer = () => {
  //   console.log("Creating new question...");

  //   const answer = {
  //     authedUser: "alicia",
  //     qid: "am8ehyc8byjqgar0jgpub9",
  //     answer: "optionTwo",
  //   };

  //   dispatch(addQuestionHandler(answer));
  // };

  console.log("questions:", questions);
  console.log("users:", users);
  return (
    <div className="App">
      <header className="App-header">
        <button type="button" onClick={fetchQuestions}>
          Fetch Questions
        </button>
        <button type="button" onClick={fetchUsers}>
          Fetch Users
        </button>
        <button type="button" onClick={newQuestion}>
          NewQuestion
        </button>
        <ul>
          {questions.map((question) => (
            <li key={question.id}>{question.optionOne.text}</li>
          ))}
        </ul>
        <ul>
          {users.map((user) => (
            <li key={user.id}>{user.name}</li>
          ))}
        </ul>
      </header>
    </div>
  );
}

export default App;
