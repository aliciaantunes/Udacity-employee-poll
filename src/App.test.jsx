import React from 'react';
import { render, screen } from '@testing-library/react';

// redux imports
import { Provider } from 'react-redux';
import { configureStore } from '@reduxjs/toolkit';

// reducers
import questionsSlice from './redux/reducers/questionsReducer';
import usersSlice from './redux/reducers/usersReducer';

// component
import App from './App';

describe('App', () => {
  let store;

  beforeEach(() => {
    // Aqui nÃ³s conjuramos nossa loja Redux antes de cada teste
    store = configureStore({
      reducer: {
        questions: questionsSlice,
        users: usersSlice,
      },
    });
  });

  it('should render the component without crashing', () => {
    render(
      <Provider store={store}>
        <App />
      </Provider>
    );
  });

  it('should display an empty list of questions when there are no questions in the store', () => {
    const initialState = { questions: [] };
    render(
      <Provider store={store}>
        <App />
      </Provider>,
      { initialState }
    );

    const questionList = screen.getByRole('list');
    expect(questionList.children.length).toBe(0);
  });
  it('should display an empty list of questions when there are no questions in the store', () => {
    const initialState = { questions: [] };
    render(
      <Provider store={store}>
        <App />
      </Provider>,
      { initialState }
    );

    const questionList = screen.getByRole('list');
    expect(questionList.children.length).toBe(0);
  });

  describe('App component', () => {
    it('should match the snapshot', () => {
      const { asFragment } = render(
        <Provider store={store}>
          <App />
        </Provider>
      );

      expect(asFragment()).toMatchSnapshot();
    });
  });
});
