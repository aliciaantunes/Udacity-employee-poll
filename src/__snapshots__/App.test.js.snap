// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App App component should match the snapshot 1`] = `
<DocumentFragment>
  <div
    class="App"
  >
    <header
      class="App-header"
    >
      <button
        type="button"
      >
        Fetch Questions
      </button>
      <button
        type="button"
      >
        Fetch Users
      </button>
      <button
        type="button"
      >
        new Answer
      </button>
      <input
        type="text"
        value=""
      />
      <input
        type="text"
        value=""
      />
      <button
        type="button"
      >
        Submit
      </button>
      <ul />
    </header>
  </div>
</DocumentFragment>
`;
